### DISCUSSION TITLE
Show Figure caption

### DISCUSSION THREAD

**Discussion by riccstick** [brechtm/rinohtype#365] [Q&A]
I have a question to the figure caption, especially how I can show for instance: "Figure 1:" in-front of the legend.

And how to show the `:numref:` from Sphinx within the PDF. Now, if I create the `numref` I have the link, however no number, just two `??` symbols.

Currently, I have set the `number_format` to `none`, because if I set it to `number`, I get an error: `paragraph.py", line 724, in number
    label.style = 'label' AttributeError: 'str' object has no attribute 'style'`.

>[figure caption]
>number_format = none

So, I am not sure how to proceed...

**Comment by brechtm** [comment]
This smells like a bug (or several). `number_format = number` should not cause an exception. In fact, it's the default, producing "Figure 1" labels for the figure captions.

Which version of rinohtype are you using?

Can you share your source files or a minimal example reproducing the issue? Then it should be easy for me to figure out what's going wrong.

**Comment by riccstick** [comment]
I tried it with the release 0.5.4 and used the documentation of rinohtype within the `doc` folder. 

So, I added a random image to a `.rst` file (e.g. intro.rst) as `figure` with a caption. 

![image](https://user-images.githubusercontent.com/44583450/194078486-567ece74-73e2-47cc-a591-732d8925075e.png)

Additionally, to show the `Figure` caption in-front, I added the `numfig = True` option to the `conf.py` for Sphinx.

![image](https://user-images.githubusercontent.com/44583450/194079156-a18f9ad4-83cb-4d11-862b-4df6778be0eb.png)

If I build it, the Sphinx documentation is working, however for the PDF I get this error: 
Exception occurred:
  File "/home/ricci/venvimpmanual/lib/python3.8/site-packages/rinoh/paragraph.py", line 724, in number
    label.style = 'label'
AttributeError: 'str' object has no attribute 'style'

Only if I change the style of the `[figure caption]` to `number_format = none` it works, however I have no Figure 1 in-front of the caption...

If I try it with the current master branch, I get the same error and for the new theme the figure caption shown two time the Fig. 1 text.
![image](https://user-images.githubusercontent.com/44583450/194082033-b4599f9c-4d29-4867-b368-4c2f41d8d73d.png)

So, yes could be a bug. :smile:

**Reply by brechtm** [reply]
Thanks for investigating! I was able to reproduce this and fix the root cause.

You can install the master version of rinohtype if you need this fix:

    pip install https://github.com/brechtm/rinohtype/archive/refs/heads/master.zip

Or as a workaround, set the label_suffix for the _figure caption_ style:

```ini
[figure caption]
label_suffix='. '
```

> If I try it with the current master branch, I get the same error and for the new theme the figure caption shown two time the Fig. 1 text.

Interesting... that must be a Sphinx or Sphinx-Immaterial bug though, since rinohtype isn't involved in the HTML build.

Note that numbering/labeling in rinohtype is independent of Sphinx, so e.g. _numfig_ shouldn't affect PDF output.

**Reply by riccstick** [reply]
Oh yeah, works nicely with the workaround. Thanks for that. 

>  Note that numbering/labeling in rinohtype is independent of Sphinx, so e.g. numfig shouldn't affect PDF output.

Ah ok thanks for the info.
